src/App.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import './App.css';
+import Counter from './components/Counter';
+import Search from './components/Search';
+
+const GENRES = ['ALL', 'DOCUMENTARY', 'COMEDY', 'HORROR', 'CRIME'];
+
+class NavBar extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { selectedGenre: 'ALL' };
+    this.updateSelectedGenre = this.updateSelectedGenre.bind(this);
+
+    this.listItems = GENRES.map((genre) =>
+      <li
+        id={genre}
+        key={genre}
+        onClick={this.updateSelectedGenre}
+        style={{ padding: '10px'}}>{genre}</li>
+    );
+  }
+
+  render() {
+    return (
+      <div>
+        <div style={{display: 'flex', listStyleType: 'none'}}>{this.listItems}</div>
+      </div>
+    );
+  }
+
+  updateSelectedGenre(event) {
+    const selectedValue = event.target.innerText;
+    this.setState({selectedGenre: selectedValue});
+    this.listItems.forEach((genre, i) => {
+      const genreTab = document.getElementById(GENRES[i]);
+      console.log(genreTab);
+      if (genreTab.innerText === selectedValue) {
+        genreTab.className =  'selected';
+      } else {
+        genreTab.className =  '';
+      }
+    })
```

Comment:
You set state, but never use it. 
And then you go through all the components manually.
It's not really React's approach. Let React do it dor you.
It's great that you already set your current tab in state.
No now let's remove `this.listItems.forEach` at all and pass classname in JSX if specific tab is selected:
```
<li className={genre === this.selectedGenre ? 'selected' : undefined} >
...
</li>
```
This is more declarative, you don't need to iterate over your elements manually and react will be able to optimize it and skip rerenders if they are not needed(in the code above react will rerender HTML on every click)

Reply:
Fixed.